CONTRIBUTING.md

```diff
@@ -57,28 +44,57 @@ Commit Message Style
 - Optionally add a long description written in full sentences beginning on the
   third line. Indent at 72 characters per line.
 
+Pull Requests
+-------------
+
+- Squash your commits into a single one before filing a pull request if
+  necessary. Each commit should represent a coherent set of changes.
+
+- Wait for the results of our CI and fix all issues. Our CI builds CAF with GCC,
+  Clang, and MSVC on various platforms and makes sure that the code is formatted
+  correctly. We also measure code coverage and require that new code is covered
+  by tests.
+
+- We require at least one code review approval before merging a pull request. We
+  take the review phase seriously and have a high bar for code quality. Multiple
+  rounds of review are common.
+
+- In case there are multiple rounds of review, please do not squash the commits
+  until everything is resolved. This makes it easier for reviewers to see what
+  has changed since the last review.
```

Comment:
Guess I was the inspiration for this :D

Reply:
No worries, happened before you as well. ;)